mate_workspace/src/joystick_module/joystick_module/joystick_publisher.py

```diff
@@ -0,0 +1,101 @@
+import pygame
+import rclpy
+from rclpy.node import Node
+
+from std_msgs.msg import Float32MultiArray
+
+
+class PublisherNode(Node):
+
+    def __init__(self):
+        super().__init__('publisher_node') # run Node constructor
+        self.publisher_ = self.create_publisher(Float32MultiArray, 'joystick_topic', 10) # publish to topic named first_topic
+        timer_period = 0.5  # seconds
+        self.timer = self.create_timer(timer_period, self.timer_callback)
+        #self.i = 0
+        pygame.init()
+        pygame.joystick.init()
+        self.joystick = pygame.joystick.Joystick(0)
+        self.joystick.init()
+
+    def timer_callback(self):
+        pygame.event.pump()
+        result= [] #Format [Left_x,Left_y,Right_x,Right_y]
+        left_stick_x = self.joystick.get_axis(0)  # Left stick horizontal
+        result.append(left_stick_x)
+        left_stick_y = self.joystick.get_axis(1)  # Left stick vertical
+        result.append(left_stick_y)
+
+        # Get the X and Y axes of the right stick
+        right_stick_x = self.joystick.get_axis(3)  # Right stick horizontal
+        result.append(right_stick_x)
+        right_stick_y = self.joystick.get_axis(4)  # Right stick vertical
+        result.append(right_stick_y)
+        msg = Float32MultiArray()
+        msg.data = result
+        self.publisher_.publish(msg)
```

Comment:
Can we write somewhere what the values in the array represent? I think that would make it easier to follow for the subscriber.

Reply:
Added comment to msg.data line as #Msg Value=[Left_x stick,Left_y stick,Right_x stick,Right_y stick] to show the array values.